staker/block_validator.go

```diff
@@ -87,9 +90,26 @@ type BlockValidatorConfig struct {
 	PendingUpgradeModuleRoot string                        `koanf:"pending-upgrade-module-root"` // TODO(magic) requires StatelessBlockValidator recreation on hot reload
 	FailureIsFatal           bool                          `koanf:"failure-is-fatal" reload:"hot"`
 	Dangerous                BlockValidatorDangerousConfig `koanf:"dangerous"`
+	MemoryFreeLimit          string                        `koanf:"memory-free-limit" reload:"hot"`
+
+	memoryFreeLimit int
 }
 
 func (c *BlockValidatorConfig) Validate() error {
+	if c.MemoryFreeLimit == "default" {
+		c.memoryFreeLimit = 1073741824 // 1GB
+		_, err := resourcemanager.NewCgroupsMemoryLimitCheckerIfSupported(c.memoryFreeLimit)
```

Comment:
I don't like the creation of an additional limit checker just to see if it's supported.

Instead - you could have in the config a boolean val "memLimitDefault" or similar, set to true if default.
Create the mem limiter only once (in NewBlockValidator as you did, or Start - both work), if it fails, look at "memLimitDefault" to choose between warn and returning an error.

Reply:
Makes sense, I'll update the impl